x-pack/plugin/downsample/src/internalClusterTest/java/org/elasticsearch/xpack/downsample/ILMDownsampleDisruptionIT.java

```diff
@@ -210,8 +210,14 @@ private void startDownsampleTaskViaIlm(
         var request = new UpdateSettingsRequest(sourceIndex).settings(
             Settings.builder().put(LifecycleSettings.LIFECYCLE_NAME, POLICY_NAME)
         );
+        // Updating index.lifecycle.name setting may fail due to the rolling restart itself,
+        // we need to attempt it in a assertBusy(...)
         assertBusy(() -> {
             try {
+                if (indexExists(sourceIndex) == false) {
```

Comment:
yes. For example the request to update index settings fails to return because the node got shutdown, but the index settings were updated. The downsampling can be kicked off. When retrying the source index could not longer exist.